### PR TITLE
feat(#452): differentiate .d.ts and .ts

### LINKED ISSUE #452: FR: differentiate `.d.ts` from `.ts` files (typescript)
A small but useful improvement would be to differentiate regular typescript files (`.ts`) from typescript declaration files (`.d.ts`)

### PR DESCRIPTION
**Pull request by chrisgrieser**
closes #452

### CODE REVIEW

File: lua/nvim-web-devicons/icons-default.lua [hunk 1]
```diff
@@ -1136,6 +1136,12 @@ local icons_by_file_extension = {
     cterm_color = "28",
     name = "D",
   },
+  ["d.ts"] = {
+    icon = "󰛦",
+    color = "#519aba",
+    cterm_color = "74",
+    name = "TypeScript Declaration",
```
Comment (alex-courtis) [inline lua/nvim-web-devicons/icons-default.lua:1143]:
```
E5248: Invalid character in group name
E5113: Error while calling lua chunk: ...pack/packer/start/chrisgrieser/lua/nvim-web-devicons.lua:296: Invalid highlight name: 'DevIconTypeScript Declaration'
stack traceback:
        [C]: in function 'nvim_set_hl'
        ...pack/packer/start/chrisgrieser/lua/nvim-web-devicons.lua:296: in function 'set_up_highlight'
        ...pack/packer/start/chrisgrieser/lua/nvim-web-devicons.lua:324: in function 'set_up_highlights'
        ...pack/packer/start/chrisgrieser/lua/nvim-web-devicons.lua:420: in function 'setup'
        ...d/site/pack/packer/start/nvim-tree.lua/lua/nvim-tree.lua:819: in function 'setup'
        /tmp/nd/nd.lua:86: in main chunk
```

vim can't have spaces in highlight group names

File: lua/nvim-web-devicons/icons-default.lua [hunk 2]
```diff
@@ -1136,6 +1136,12 @@ local icons_by_file_extension = {
     cterm_color = "28",
     name = "D",
   },
+  ["d.ts"] = {
+    icon = "󰛦",
```
Comment (alex-courtis) [inline lua/nvim-web-devicons/icons-default.lua:1140]: This one's material design, the other javascript icons are seti.

Please change to seti for consistency.

You can use https://www.nerdfonts.com/cheat-sheet and paste the icon into the search box.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Looks good, a couple of changes needed.

File: lua/nvim-web-devicons/icons-default.lua (hunk 1 above)
Comment (chrisgrieser) [reply lua/nvim-web-devicons/icons-default.lua]: fixed

**Comment by chrisgrieser** [comment]
all fixed

**Review by alex-courtis — APPROVED** [review]

**Review by alex-courtis — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**